File: daily/rust/518_coin_change_II.rs
impl Solution {
    pub fn change(amount: i32, coins: Vec<i32>) -> i32 {
        let total = amount as usize;
        let mut dp: Vec<i32> = vec![0; total + 1];
        dp[0] = 1;
        for &coin in coins.iter() {
            let coin_val = coin as usize;
            for i in coin_val..=total {
                dp[i] += dp[i - coin_val];
            }
        }
        dp[total]
    }
}

File: daily/cpp/518_coin_change_II.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
 public:
  int change(int amount, vector<int>& coins) {
    vector<int> dp(amount + 1);
    for (int& coin : coins) {
      for (int i = coin; i <= amount; ++i) {
        dp[i] += dp[i - coin];
      }
    }
    return dp[amount];
  }
};